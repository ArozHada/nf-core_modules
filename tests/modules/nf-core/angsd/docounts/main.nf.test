nextflow_process {

    name "Test Process ANGSD_DOCOUNTS"
    script "modules/nf-core/angsd/docounts/main.nf"
    process "ANGSD_DOCOUNTS"
    config "tests/modules/nf-core/angsd/docounts/nextflow.config"
    tag "angsd_docounts"

    test("Human Illumina PE Sorted BAM") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file(params.test_data['homo_sapiens']['illumina']['test_paired_end_markduplicates_sorted_bam'], checkIfExists: true),
                            file(params.test_data['homo_sapiens']['illumina']['test_paired_end_markduplicates_sorted_bam_bai'], checkIfExists: true),
                            []
                        ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }

    }

}
