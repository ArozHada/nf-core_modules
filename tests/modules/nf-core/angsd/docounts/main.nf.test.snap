{
    "Human Illumina PE Sorted BAM": {
        "content": [
            [
                "test.counts.gz:md5,c1d296252105cce5f4a891985d98f3db",
                "test.icnts.gz:md5,de524782a29d647d4a358fb98756665b",
                "test.pos.gz:md5,2ac40e7f6bf7e5735c8267b892b99368",
                "versions.yml:md5,211a931297225fefaa891d02b755e215"
            ]
        ],
        "timestamp": "2023-04-06T01:18:44+0000"
    }
}